### PR TITLE
add provision to delete kueue resources created during test

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
# Issue link

[RHOAIENG-7684](https://issues.redhat.com/browse/RHOAIENG-7684)

# What changes have been made

- Added support method to delete kueue resources
- Updated kueue resource creation flow to create resources using unique name for avoiding naming conflicts and to execute tests parallely
- Updated e2e + upgrade tests to delete kueue resources created during test execution.

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: tests/e2e/support.py [hunk 1]
```diff
@@ -163,3 +163,31 @@ def create_kueue_resources(
             body=local_queue_json,
         )
         print(f"'{local_queue}' created in namespace '{self.namespace}'")
+
+
+def delete_kueue_resources(
+    self, cluster_queue="cluster-queue-mnist", flavor="default-flavor-mnist"
```
Comment (sutaakar) [inline tests/e2e/support.py:169]: Can you make those names global variables?
That way they will be kept in sync with creation functions.

File: tests/e2e/support.py [hunk 2]
```diff
@@ -163,3 +163,31 @@ def create_kueue_resources(
...
+            version="v1beta1",
+            name=flavor,
+        )
+        print(f"'{flavor}' resource-flavor deleted")
+    except Exception as e:
+        print(f"Error deleting resource-flavor '{flavor}' : {e}")
+
+    # Delete if given cluster-queue exists
+    try:
+        self.custom_api.delete_cluster_custom_object(
```
Comment (astefanutti) [inline tests/e2e/support.py:185]: Help me remember: on behalf of which user these operations are performed?

Comment (sutaakar) [reply tests/e2e/support.py:185]: I suppose it is https://github.com/project-codeflare/codeflare-sdk/blob/main/.github/workflows/e2e_tests.yaml#L109-L125

Comment (astefanutti) [reply tests/e2e/support.py:185]: Right. Ideally there would be two users impersonating the two different personas, SDK user on one hand, RHOAI admin on the other hand, and the latter would perform operations related to queue management.

**Review by sutaakar — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]
LGTM, Thanks

**Review by jiripetrlik — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]

**Comment by Srihari1192** [comment]
@abhijeet-dhumal Can you add kueue deletion for [appwrapper ](https://github.com/abhijeet-dhumal/codeflare-sdk/blob/add-kueue-deletion/tests/e2e/mnist_raycluster_sdk_aw_kind_test.py) tests too

**Pull request merged**